node/pkg/nodeinit/calico-init_linux.go

```diff
@@ -66,10 +72,28 @@ func ensureBPFFilesystem() error {
 
 	// If we get here, the BPF filesystem is not mounted.  Try to mount it.
 	logrus.Info("BPF filesystem is not mounted.  Trying to mount it...")
-	err = syscall.Mount(bpfFSMountPoint, bpfFSMountPoint, "bpf", 0, "")
+	err = syscall.Mount(bpf.DefaultBPFfsPath, bpf.DefaultBPFfsPath, "bpf", 0, "")
 	if err != nil {
 		return fmt.Errorf("failed to mount BPF filesystem: %w", err)
 	}
 	logrus.Info("Mounted BPF filesystem.")
 	return nil
 }
+
+// ensureCgroupV2Filesystem() enters the cgroup and mount namespace of the process
+// with PID 1 running on a host to allow felix running in calico-node access the root of cgroup namespace.
+// This is needed by felix to attach CTLB programs and implement k8s services correctly.
+func ensureCgroupV2Filesystem() error {
+	mountCmd := exec.Command(
+		"nsenter", "--cgroup=/node-proc/1/ns/cgroup", "--mount=/node-proc/1/ns/mnt",
+		"mount", "-t", "cgroup2", "none", bpf.CgroupV2Path)
```

Comment:
Did you figure out if the version of  `mount` that gets executed is the one inside the container or outside?  If it's outside then we wouldn't even need to ship `mount` in the container.

Reply:
it's the mount from host that get executed. Will remove `mount` binary from calico-node